Source/Core/VideoBackends/Software/Clipper.cpp

```diff
@@ -388,19 +404,67 @@ void ProcessLine(OutputVertexData* lineV0, OutputVertexData* lineV1)
 
     OutputVertexData triangle[3];
 
-    CopyVertex(&triangle[0], v0, screenDx, screenDy, 0);
-    CopyVertex(&triangle[1], v1, screenDx, screenDy, 0);
-    CopyVertex(&triangle[2], v1, -screenDx, -screenDy, bpmem.lineptwidth.lineoff);
+    CopyLineVertex(&triangle[0], v0, screenDx, screenDy, 0);
+    CopyLineVertex(&triangle[1], v1, screenDx, screenDy, 0);
+    CopyLineVertex(&triangle[2], v1, -screenDx, -screenDy, bpmem.lineptwidth.lineoff);
 
     // ccw winding
     Rasterizer::DrawTriangleFrontFace(&triangle[2], &triangle[1], &triangle[0]);
 
-    CopyVertex(&triangle[1], v0, -screenDx, -screenDy, bpmem.lineptwidth.lineoff);
+    CopyLineVertex(&triangle[1], v0, -screenDx, -screenDy, bpmem.lineptwidth.lineoff);
 
     Rasterizer::DrawTriangleFrontFace(&triangle[0], &triangle[1], &triangle[2]);
   }
 }
 
+static void CopyPointVertex(OutputVertexData* dst, const OutputVertexData* src, float offset,
+                            u32 sOffset, bool px, bool py)
+{
+  dst->projectedPosition = src->projectedPosition;
+  dst->screenPosition.x = src->screenPosition.x + (px ? 1 : -1) * offset;
+  dst->screenPosition.y = src->screenPosition.y + (py ? 1 : -1) * offset;
+  dst->screenPosition.z = src->screenPosition.z;
+
+  dst->normal = src->normal;
+  dst->color = src->color;
+
+  dst->texCoords = src->texCoords;
+
+  if (LINE_PT_TEX_OFFSETS[sOffset] != 0)
+  {
+    for (u32 coordNum = 0; coordNum < xfmem.numTexGen.numTexGens; coordNum++)
+    {
+      if (bpmem.texcoords[coordNum].s.point_offset)
+      {
+        if (px)
```

Comment:
We brace if statements where the body travels more than one line , ditto for the `if (py)` if statement below this one.